orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -214,10 +213,13 @@ public static Schema convert(TypeDescription orcSchema) {
         "Error in ORC file, children fields and names do not match.");
 
     List<Types.NestedField> icebergFields = Lists.newArrayListWithExpectedSize(children.size());
-    AtomicInteger lastColumnId = new AtomicInteger(getMaxIcebergId(orcSchema));
-    for (int i = 0; i < children.size(); i++) {
-      icebergFields.add(convertOrcToIceberg(children.get(i), childrenNames.get(i),
-          lastColumnId::incrementAndGet));
+    OrcToIcebergVisitor schemaConverter = new OrcToIcebergVisitor(icebergToOrcMapping("root", orcSchema));
+    for (TypeDescription child : orcSchema.getChildren()) {
+      OrcToIcebergVisitor.visit(child, schemaConverter).ifPresent(icebergFields::add);
+    }
+
+    if (icebergFields.size() == 0) {
+      throw new IllegalArgumentException("ORC schema has no Iceberg mappings");
```

Comment:
I wouldn't refer to the ID annotations as mappings. How about "ORC schema does not contain Iceberg IDs"?